### PR TITLE
Update versions for dependencies to fix CVEs

### PR DESCRIPTION
**Pull request by spaparaju**
Update versions for dependencies to fix CVEs.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -31,7 +31,7 @@ flask-restful==0.3.7
 flask-security==3.0.0
 flask-sqlalchemy==2.3.2
 flask-wtf==0.14.2         # via flask-security
-flask==0.12.4
+flask>=1.0.0
```
Comment (samuzzal-choudhury) [inline requirements.txt]: @SriKrishnaPaparaju requirements.txt needs a resolved version. So, please replace `>=` with `==`.

File: requirements.txt [hunk 2]
```diff
@@ -60,7 +60,7 @@ six==1.12.0               # via cryptography, flask-appconfig, flask-cors, flask
 smmap2==2.0.5             # via gitdb2
 speaklater==1.3           # via flask-babelex
 sqlalchemy==1.3.2         # via flask-sqlalchemy
-urllib3==1.24.1           # via requests
+urllib3>=1.24.2           # via requests
```
Comment (samuzzal-choudhury) [inline requirements.txt]: @SriKrishnaPaparaju requirements.txt will need a resolved version. So, please replace `>=` with `==`.

File: tests/requirements.txt [hunk 3]
```diff
@@ -12,7 +12,7 @@ flexmock==0.10.2
 freezegun==0.3.7
 isort==4.2.5              # via pylint
 itsdangerous==0.24        # via flask
-Jinja2==2.8               # via flask
+Jinja2>=2.10.1               # via flask
```
Comment (samuzzal-choudhury) [inline tests/requirements.txt]: @SriKrishnaPaparaju requirements.txt will need a resolved version. So, please replace `>=` with `==`.

File: tests/requirements.txt [hunk 4]
```diff
@@ -23,7 +23,7 @@ pytest-flask==0.10.0
 pytest==3.0.3
 pytest-cov==2.5.1
 python-dateutil==2.5.3    # via freezegun
-requests==2.11.1
+requests>=2.20.0
```
Comment (samuzzal-choudhury) [inline tests/requirements.txt]: @SriKrishnaPaparaju requirements.txt will need a resolved version. So, please replace `>=` with `==`. Also, there is flask dependency in this file which has to be updated with the version `==1.0.0`.

File: tests/requirements.txt (hunk 3 above)
Comment (SriKrishnaPaparaju) [reply tests/requirements.txt]: Changed to resolved versions

**Comment by samuzzal-choudhury** [comment]
[test]

**Comment by tisnik** [comment]
It's needed to bump-up versions in `requirements.in` and then run
`pip-compile --output-file requirements.txt requirements.in` as described in `requirements.txt`

**Comment by SriKrishnaPaparaju** [comment]
> Please bump-up versions in the file `requirements.in`.
> 
> Then the file `requirements.txt` needs to be generated by:
> 
> ```
> pip-compile --output-file requirements.txt requirements.in
> ```
> 
> (additionally it is good to check `git diff requirements.txt` whether automagically created changes make sense)
> 
> Thanks a lot in advance @SriKrishnaPaparaju

Thanks @tisnik . I guess these version changes require some code level changes as 'flask_appconfig' (last release was in 2015) being used which is pinning the version of ''flask'

**Comment by samuzzal-choudhury** [comment]
[test]

**Comment by miteshvp** [comment]
> I guess these version changes require some code level changes as 'flask_appconfig' (last release was in 2015) being used which is pinning the version of ''flask'

@SriKrishnaPaparaju - those versions are being picked from your global installs. One way to get the latest is to run
`pip-compile --upgrade --output-file requirements.txt requirements.in` and it will upgrade all those deps to the latest ones.

**Comment by SriKrishnaPaparaju** [comment]
[test]

**Pull request closed without merging**